rust/benches/array_from_vec.rs

```diff
@@ -37,7 +35,7 @@ fn array_from_vec(n: usize) {
     let arr_data = ArrayDataBuilder::new(DataType::Int32)
         .add_buffer(Buffer::from(v))
         .build();
-    let _: PrimitiveArray<i32> = PrimitiveArray::from(Rc::new(arr_data));
+    criterion::black_box(PrimitiveArray::<i32>::from(arr_data));
```

Comment:
Just as a side note:
It's completely fine to use this to compare one change to another (like "did this PR result in any regression"), but it may not work to compare the rust against the C++ implementation, because the black box is not probably implemented under stable rust (see [criterion source code](https://github.com/japaric/criterion.rs/blob/master/src/lib.rs#L129-L130) and [the reason why `black_box` is unstable](https://github.com/nikomatsakis/rust-memory-model/issues/45)).